accord-core/src/main/java/accord/coordinate/Invalidate.java

```diff
@@ -68,15 +72,17 @@ private Invalidate(Node node, Shard shard, Ballot ballot, TxnId txnId, RoutingKe
         this.informKeys = informKeys;
         this.invalidateWithKey = invalidateWithKey;
         this.recoverIfAtLeast = recoverIfAtLeast;
-        this.preacceptTracker = new QuorumShardTracker(shard);
+        this.tracker = new QuorumTracker.QuorumShardTracker(shard);
     }
 
-    public static Invalidate invalidateIfNotWitnessed(Node node, TxnId txnId, RoutingKeys informKeys, RoutingKey invalidateWithKey, BiConsumer<Outcome, Throwable> callback)
+    public static Invalidate invalidate(Node node, TxnId txnId, RoutingKeys informKeys, RoutingKey invalidateWithKey, BiConsumer<Outcome, Throwable> callback)
     {
         return invalidate(node, txnId, informKeys, invalidateWithKey, PreAccepted, callback);
     }
 
-    public static Invalidate invalidate(Node node, TxnId txnId, RoutingKeys informKeys, RoutingKey invalidateWithKey, BiConsumer<Outcome, Throwable> callback)
+    // TODO (now): (file separately) this is a bug, as it is possible for a preaccept to still be in-flight.
+    //             It's not even safe to do it after an initial Invalidate round unless we ensure that recovery never pre-accepts with txnId (which we should also enforce)
```

Comment:
Take a look at my next patch. But the problem here is the `invalidateWithKey` which is only safe is we have prevented a pre-accept from taking place with our ballot, otherwise a pre-accept might have been inflight and reached there between when we started the invalidation and our BeginInvalidation arrives.